cunumeric/module.py

```diff
@@ -6227,3 +6227,719 @@ def bincount(
+    pos = q * (n - 1)
+
+    # k = floor(pos)
+    # gamma = 1.0 if pos - k >= 0.5 else 0.0
+
+    k = np.round(pos)
+    gamma = 0.0
+
+    j = int(k)
+    return (gamma, j)
+
+
+# helper: test array if sorted:
+#
+def is_sorted(a):
+    return all(a[:-1] <= a[1:])
+
+
+# helper: test if array inputs are different:
+#
+def is_diff(arr1, arr2):
+    return len(arr1) != len(arr2) or any(
+        [arr1[i] != arr2[i] for i in range(0, len(arr1))]
+    )
+
+
+# for the case when axis = tuple (non-singleton)
+# reshuffling might have to be done (if tuple is non-consecutive)
+# and the src array must be collapsed along that set of axes
+#
+# args:
+#
+# arr:    [in] source nd-array on which quantiles are calculated;
+# axes_set: [in] tuple or list of axes (indices less than arr dimension);
+#
+# return: pair: (minimal_index, reshuffled_and_collapsed source array)
+# TODO: check if reshuffling, reshaping is done in-place!
+#
+def reshuffle_reshape(arr, axes_set):
+    ndim = len(arr.shape)
+
+    if not is_sorted(axes_set):
+        sorted_axes = tuple(sort(axes_set))
+    else:
+        sorted_axes = tuple(axes_set)
+
+    min_dim_index = sorted_axes[0]
+    num_axes = len(sorted_axes)
+    reshuffled_axes = tuple(range(min_dim_index, min_dim_index + num_axes))
+
+    non_consecutive = is_diff(sorted_axes, reshuffled_axes)
+    if non_consecutive:
+        arr_shuffled = moveaxis(arr, sorted_axes, reshuffled_axes)
+    else:
+        arr_shuffled = arr
+
+    # shape_reshuffled = arr_shuffled.shape # debug
+    collapsed_shape = np.product(
+        [arr_shuffled.shape[i] for i in reshuffled_axes]
+    )
+
+    redimed = tuple(range(0, min_dim_index + 1)) + tuple(
+        range(min_dim_index + num_axes, ndim)
+    )
+    reshaped = tuple(
+        [
+            collapsed_shape if k == min_dim_index else arr_shuffled.shape[k]
+            for k in redimed
+        ]
+    )
+
+    arr_reshaped = arr_shuffled.reshape(reshaped)
+    return (min_dim_index, arr_reshaped)
+
+
+# args:
+#
+# arr:      [in] source nd-array on which quantiles are calculated;
+#                preccondition: assumed sorted!
+# q_arr:    [in] quantile input values nd-array;
+# axis:     [in] axis along which quantiles are calculated;
+# method:   [in] func(q, n) returning (gamma, j),
+#                where = array1D.size;
+# keepdims: [in] boolean flag specifying whether collapsed axis
+#                should be kept as dim=1;
+# to_dtype: [in] dtype to convert the result to;
+# qs_all:   [in/out] result pass through or created (returned)
+# input_is_scalar: [in] specifies if q input was scalar
+# return: nd-array of quantile output values
+#         where its shape is obtained as:
+#         concatenating q_arr.shape with arr.shape \ {axis}
+#         (the shape of `arr` obtained by taking the `axis` out)
+#
+def quantile_impl(
+    arr,
+    q_arr,
+    axis,
+    axes_set,
+    original_shape,
+    method,
+    keepdims,
+    to_dtype,
+    qs_all,
+    input_is_scalar,
```

Comment:
Looks like this argument isn't used anymore

Reply:
Removed.